### PR TITLE
Login/out: add spacing support

### PR DESCRIPTION
**Pull request by carolinan**
## What?

Adds margin and padding support to the login/out block.

## Why?

For consistency. Part of https://github.com/WordPress/gutenberg/pull/45147.

## How?

The PR adds spacing support to block.json.

## Testing Instructions

Make sure that the active theme supports spacing, for example, in theme.json.
Create a new post or page.
Add login/out block.
Confirm that there is a dimension panel, and that the margin and padding settings are available but not enabled by default.
Test the margin and padding settings in the editor and front.

-Go to the Site Editor, add a login/out block, and change the margin and padding via the block settings in the Styles sidebar.
-Adjust the spacing and margin via theme.json.

Example theme.json code: add this to styles > blocks:

```
"core/loginout": {
				"spacing": {
					"padding": {
						"top": "15px",
						"right": "15px",
						"bottom": "15px",
						"left": "15px"
					},
					"margin": {
						"top": "150px",
						"right": "15px",
						"bottom": "150px",
						"left": "15px"
					}
				}
			},
```

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/7422055/8b985ecc-87fe-44e4-8b74-5d34c1223007

### CODE REVIEW

**Comment by carolinan** [comment]
Left and right margin isn't doing anything in the block editor if "Inner blocks use content width" is enabled. How should we handle it? Only enable vertical margin?

File: packages/block-library/src/loginout/block.json [hunk 1]
```diff
@@ -19,6 +19,10 @@
 	},
 	"supports": {
 		"className": true,
+		"spacing": {
+			"margin": true,
+			"padding": true
+		},
```
Comment (aaronrobertshaw) [inline packages/block-library/src/loginout/block.json]:
```suggestion
		"spacing": {
			"margin": true,
			"padding": true,
			"__experimentalDefaultControls": {
				"margin": false,
				"padding": false
			}
		},
```

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for continuing to chip away at design tool adoption @carolinan 👍 

> Left and right margin isn't doing anything in the block editor if "Inner blocks use content width" is enabled. How should we handle it? Only enable vertical margin?

Other blocks still adopt support for horizontal margins as individual blocks can be nested within a container block. In that case, the individual block isn't constrained by layout styles.

With regards to this PR, the spacing support styles for this block are working as I'd expect them to. 

There's one small tweak I think needs to be made though. With  https://github.com/WordPress/gutenberg/pull/52256 blocks such as this one are having their dimension controls hidden by default. I think we should do the same here.

I've left the suggested tweak as an inline comment.

**Review by aaronrobertshaw — APPROVED** [review]
LGTM and is working as advertised 👍 

The spacing controls are now hidden by default and match other blocks.

Thanks for your patience on this @carolinan 🙇

**Comment by carolinan** [comment]
Got to catch them all... I mean, fix them all ;) Even if it takes time.

**Pull request merged**